sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -699,23 +692,24 @@ ProgramManager::getOrCreateKernel(const ContextImplPtr &ContextImpl,
 
   KernelProgramCache &Cache = ContextImpl->getKernelProgramCache();
 
-  std::string CompileOpts, LinkOpts;
   SerializedObj SpecConsts;
+  std::string_view CompileOpts, LinkOpts;
   if (Prg) {
```

Comment:
I think we can remove `program_impl *` from here as well. When I look at the callers of `getOrCreateKernel` that pass a non-null `program_impl *`, that call is always guarded by `program_impl::is_cacheable()`, which is only true for programs built with `program_impl::build_with_kernel_name` and that has no callers.